File: IOS/v2/iosx/fragment/app/FragmentManagerViewModel.swift
package iOSx.fragment.app;.swift
import iOS.util.Log;.swift
import java.util.ArrayList;.swift
import java.util.HashMap;.swift
import java.util.Iterator;.swift
final class FragmentManagerViewModel extends yag {.swift
public static final d i = new Object();.swift
public final HashMap c = new HashMap();.swift
public final HashMap d = new HashMap();.swift
public final HashMap e = new HashMap();.swift
public final boolean f;.swift
public boolean g = false;.swift
public boolean h = false;.swift
public FragmentManagerViewModel(boolean z) {.swift
this.f = z;.swift
}.swift
public final boolean equals(Object obj) {.swift
if (this == obj) {.swift
return true;.swift
}.swift
if (obj == null || FragmentManagerViewModel.class = obj.getClass()) {.swift
return false;.swift
}.swift
FragmentManagerViewModel fragmentManagerViewModel = (FragmentManagerViewModel) obj;.swift
return this.c.equals(fragmentManagerViewModel.c) && this.d.equals(fragmentManagerViewModel.d) && this.e.equals(fragmentManagerViewModel.e);.swift
}.swift
public final int hashCode() {.swift
int hashCode = this.d.hashCode();.swift
return this.e.hashCode() + ((hashCode + (this.c.hashCode() * 31)) * 31);.swift
}.swift
public final void i() {.swift
if (Log.isLoggable("FragmentManager", 3)) {.swift
toString();.swift
}.swift
this.g = true;.swift
}.swift
public final void j(a aVar) {.swift
if (this.h) {.swift
HashMap hashMap = this.c;.swift
if (hashMap.containsKey(aVar.w)) {.swift
hashMap.put(aVar.w, aVar);.swift
if (Log.isLoggable("FragmentManager", 2)) {.swift
aVar.toString();.swift
}.swift
}.swift
}.swift
}.swift
public final void k(String str, boolean z) {.swift
HashMap hashMap = this.d;.swift
FragmentManagerViewModel fragmentManagerViewModel = (FragmentManagerViewModel) hashMap.get(str);.swift
if (fragmentManagerViewModel = null) {.swift
if (z) {.swift
ArrayList arrayList = new ArrayList();.swift
arrayList.addAll(fragmentManagerViewModel.d.keySet());.swift
Iterator it = arrayList.iterator();.swift
while (it.hasNext()) {.swift
fragmentManagerViewModel.k((String) it.next(), true);.swift
}.swift
}.swift
fragmentManagerViewModel.i();.swift
hashMap.remove(str);.swift
}.swift
HashMap hashMap2 = this.e;.swift
gbg gbg = (gbg) hashMap2.get(str);.swift
if (gbg = null) {.swift
gbg.a();.swift
hashMap2.remove(str);.swift
}.swift
}.swift
public final void l(a aVar) {.swift
if (= null && Log.isLoggable("FragmentManager", 2)) {.swift
aVar.toString();.swift
}.swift
}.swift
public final String toString() {.swift
StringBuilder sb = new StringBuilder("FragmentManagerViewModel{");.swift
sb.append(Integer.toHexString(System.identityHashCode(this)));.swift
sb.append("} Fragments (");.swift
Iterator it = this.c.values().iterator();.swift
while (it.hasNext()) {.swift
sb.append(it.next());.swift
if (it.hasNext()) {.swift
sb.append(", ");.swift
}.swift
}.swift
sb.append(") Child Non Config (");.swift
Iterator it2 = this.d.keySet().iterator();.swift
while (it2.hasNext()) {.swift
sb.append((String) it2.next());.swift
if (it2.hasNext()) {.swift
sb.append(", ");.swift
}.swift
}.swift
sb.append(") ViewModelStores (");.swift
Iterator it3 = this.e.keySet().iterator();.swift
while (it3.hasNext()) {.swift
sb.append((String) it3.next());.swift
if (it3.hasNext()) {.swift
sb.append(", ");.swift
}.swift
}.swift
sb.append(')');.swift
return sb.toString();.swift
}.swift
}.swift

File: Android/sources/androidx/fragment/app/FragmentManagerViewModel.java
package androidx.fragment.app;

import android.util.Log;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Iterator;

final class FragmentManagerViewModel extends yag {
    public static final d i = new Object();
    public final HashMap c = new HashMap();
    public final HashMap d = new HashMap();
    public final HashMap e = new HashMap();
    public final boolean f;
    public boolean g = false;
    public boolean h = false;

    public FragmentManagerViewModel(boolean z) {
        this.f = z;
    }

    public final boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || FragmentManagerViewModel.class != obj.getClass()) {
            return false;
        }
        FragmentManagerViewModel fragmentManagerViewModel = (FragmentManagerViewModel) obj;
        return this.c.equals(fragmentManagerViewModel.c) && this.d.equals(fragmentManagerViewModel.d) && this.e.equals(fragmentManagerViewModel.e);
    }

    public final int hashCode() {
        int hashCode = this.d.hashCode();
        return this.e.hashCode() + ((hashCode + (this.c.hashCode() * 31)) * 31);
    }

    public final void i() {
        if (Log.isLoggable("FragmentManager", 3)) {
            toString();
        }
        this.g = true;
    }

    public final void j(a aVar) {
        if (!this.h) {
            HashMap hashMap = this.c;
            if (!hashMap.containsKey(aVar.w)) {
                hashMap.put(aVar.w, aVar);
                if (Log.isLoggable("FragmentManager", 2)) {
                    aVar.toString();
                }
            }
        }
    }

    public final void k(String str, boolean z) {
        HashMap hashMap = this.d;
        FragmentManagerViewModel fragmentManagerViewModel = (FragmentManagerViewModel) hashMap.get(str);
        if (fragmentManagerViewModel != null) {
            if (z) {
                ArrayList arrayList = new ArrayList();
                arrayList.addAll(fragmentManagerViewModel.d.keySet());
                Iterator it = arrayList.iterator();
                while (it.hasNext()) {
                    fragmentManagerViewModel.k((String) it.next(), true);
                }
            }
            fragmentManagerViewModel.i();
            hashMap.remove(str);
        }
        HashMap hashMap2 = this.e;
        gbg gbg = (gbg) hashMap2.get(str);
        if (gbg != null) {
            gbg.a();
            hashMap2.remove(str);
        }
    }

    public final void l(a aVar) {
        if (!this.h && this.c.remove(aVar.w) != null && Log.isLoggable("FragmentManager", 2)) {
            aVar.toString();
        }
    }

    public final String toString() {
        StringBuilder sb = new StringBuilder("FragmentManagerViewModel{");
        sb.append(Integer.toHexString(System.identityHashCode(this)));
        sb.append("} Fragments (");
        Iterator it = this.c.values().iterator();
        while (it.hasNext()) {
            sb.append(it.next());
            if (it.hasNext()) {
                sb.append(", ");
            }
        }
        sb.append(") Child Non Config (");
        Iterator it2 = this.d.keySet().iterator();
        while (it2.hasNext()) {
            sb.append((String) it2.next());
            if (it2.hasNext()) {
                sb.append(", ");
            }
        }
        sb.append(") ViewModelStores (");
        Iterator it3 = this.e.keySet().iterator();
        while (it3.hasNext()) {
            sb.append((String) it3.next());
            if (it3.hasNext()) {
                sb.append(", ");
            }
        }
        sb.append(')');
        return sb.toString();
    }
}
